_vector-search/ai-search/building-flows.md

```diff
@@ -0,0 +1,443 @@
+                "k": 10
+            }
+        }
+    }
+}
+```
+
+{% include copy.html %}
+
+---
+
+## Hybrid search
+
+Hybrid search combines keyword and vector search. This example demonstrates how to configure hybrid search.
+
+### ML resources
+
+Create and deploy an [Amazon Titan Text Embeddings model on Amazon Bedrock](https://github.com/opensearch-project/dashboards-flow-framework/blob/main/documentation/models.md#amazon-bedrock-titan-text-embedding).
+
+### Index
+
+Ensure that the index settings include `index.knn: true` and that your index contains a `knn_vector` field specified in the mappings, as follows:
+
+```json
+{
+  "settings": {
+    "index": {
+      "knn": true
+    }
+  },
+  "mappings": {
+    "properties": {
+      "<embedding_field_name>": {
+        "type": "knn_vector",
+        "dimension": "<embedding_size>"
+      }
+    }
+  }
+}
+```
+
+{% include copy.html %}
+
+### Ingest pipeline
+
+Configure a single ML inference processor. Map your input text to the `inputText` model input field. Optionally, map the output `embedding` to a new document field.
+
+### Search pipeline
+
+Configure an ML inference search request processor and a normalization processor.
+
+**For the ML inference processor**, map the query field containing the input text to the `inputText` model input field. Optionally, map the output `embedding` to a new field. Override the query so that it contains a `hybrid` query. Make sure to specify the `embedding_field`, `text_field`, and `text_field_input`:
+
+```json
+{
+    "_source": {
+        "excludes": [
+            "<embedding_field>"
+        ]
+    },
+    "query": {
+        "hybrid": {
+            "queries": [
+                {
+                    "match": {
+                        "<text_field>": {
+                            "query": "<text_field_input>"
+                        }
+                    }
+                },
+                {
+                    "knn": {
+                        "<embedding_field>": {
+                            "vector": ${embedding},
+                            "k": 10
+                        }
+                    }
+                }
+            ]
+        }
+    }
+}
+```
+
+{% include copy.html %}
+
+**For the normalization processor**, configure weights for each subquery. For more information, see the [hybrid search normalization processor example]({{site.url}}{{site.baseurl}}/search-plugins/hybrid-search/#step-3-configure-a-search-pipeline).
+
+---
+
+## Basic RAG (document summarization)
+
+This example demonstrates how to configure basic retrieval-augmented generation (RAG).
+
+The following example shows a simplified connector blueprint for the [Claude v1 messages API](https://docs.anthropic.com/en/api/getting-started#examples). While connector blueprints and model interfaces may evolve over time, this example demonstrates how to abstract complex API interactions into a single `prompt` field input.
```

Comment:
At the link, only the "Anthropic API" is referenced. Please confirm the API name.